### DISCUSSION TITLE
[Impersonating Authenticated User] Checking Row-Level-Security auth.uid() doesn't work

### DISCUSSION THREAD

**Discussion by luisedelgado** [supabase/supabase#23161] [Questions]
Hi all! In setting up my project's database, I've basically written a few tables, added dummy data, and all rows have the same RLS policy: **-Authenticated users can only see their own data-**. Now, what I'm seeing is that when I look at my tables' rows in the GUI while impersonating a postgres role, I can see all my data. But after adding my personal email as a verified user, and impersonating that signed-in email, I can't see anything. I'm new to both Supabase, as well as Postgres so I'm not sure if I'm missing something obvious.

Below is the `SELECT` policy that I'm setting on all tables (I have not set any policies for any other operation as I just want read access). `representative_id` is present in all tables, and contains the **user_id** that I get back when signing in through the Python client. I've also tried fetching data through the Python client after having signed-in with my personal email but I'm also not getting anything back. I've seen [other posts](https://github.com/orgs/supabase/discussions/6592) where they mention that Server talks anonymously to Supabase via the `anon` key, and therefore we need to overwrite it with the current user's `access_token`, but haven't found that the Python client's APIs allow for such thing. Any idea what could be happening? Thanks in advance for the help!

```
alter policy "Enable read access for authenticated users"
on "public"."reports"
to authenticated
using (
  (auth.uid() = representative_id)
);

```
<img width="702" alt="Screenshot 2024-04-22 at 5 08 03 PM" src="https://github.com/supabase/supabase/assets/16198573/894fcb0d-554f-4d76-9c6d-300a36fd8c08">

**Comment by GaryAustin1** [comment]
The UI seems to be working correctly.  
![image](https://github.com/supabase/supabase/assets/54564956/2f91aa9d-6234-49e4-a8d7-dbef2f7c46a5)

Double check your user id's in the table match the uuid of your email picked user in the authentication tab.

**Reply by luisedelgado** [reply]
I've been **very** thorough in double-checking, I made sure there weren't any leading or trailing spaces too. The strings match perfectly :(. What would be the best route to get support by someone looking at this with me? Does Supabase have developer-specific Support? I feel I'm incredibly close to getting it working, but something is just not working

**Reply by GaryAustin1** [reply]
Not that I know of.  Normally here and Discord (usually more interactive) with other users is it.
You probably need to show your actual table row with column names/types.  Is representative_id a UUID column or a string for instance, auth.uid() is UUID.    RLS is silent when it fails on selects.

I just hard refreshed my browser and UI still works as expected, so don't think it is the UI.

**Reply by luisedelgado** [reply]
Ok I just realized what was happening. I was setting my RLS policies as Restrictive instead of Permissive. That section in the GUI probably deserves a bit of a friendlier explanation to the user. Thanks anyway!

**Reply by GaryAustin1** [reply]
Ah... your policy you showed did not use restrictive 
![image](https://github.com/supabase/supabase/assets/54564956/b66bd4f5-b4d3-43d1-9f19-cf2e14a9dc28)
As default is permissive.

If you have feedback for Supabase on better UI/Docs I recommend you post it supabase/supabase issue here.  They are unlikely to see this at this point.
